File: src/main/java/com/silverboxsoft/dynamodbtool/controller/inputdialog/DynamoDbStringSetInputDialog.java
package com.silverboxsoft.dynamodbtool.controller.inputdialog;

import java.util.List;

import javafx.scene.Node;
import javafx.scene.control.TextField;

public class DynamoDbStringSetInputDialog extends AbsDynamoDbSetInputDialog<String> {

	public DynamoDbStringSetInputDialog(List<String> dynamoDbRecord, String dialogTitle) {
		super(dynamoDbRecord, dialogTitle);
	}

	@Override
	protected String getTypeString() {
		return "STRING";
	};

	@Override
	protected Node getAttrubuteBox(int recIndex, String attr) {
		TextField textField = new TextField();
		textField.setText(attr);
		return textField;
	}

	@Override
	protected String getCurrentAttrubuteValue(Node valueNode) {
		TextField valField = (TextField) valueNode;
		return valField.getText();
	}

	@Override
	protected String getInitAttribute() {
		return "";
	}

	@Override
	protected boolean isSameValue(String valA, String valB) {
		return valA.equals(valB);
	}
}

File: src/main/java/com/silverboxsoft/dynamodbtool/controller/inputdialog/DynamoDbStringSetInputDialog.kt
package com.silverboxsoft.dynamodbtool.controller.inputdialog

import com.silverboxsoft.dynamodbtool.utils.DynamoDbUtils
import javafx.scene.Node
import javafx.scene.control.*

class DynamoDbStringSetInputDialog(dynamoDbRecord: List<String>, dialogTitle: String) : AbsDynamoDbSetInputDialog<String>(dynamoDbRecord, dialogTitle) {
    override val typeString: String
        get() = "STRING"

    override fun getAttributeBox(recIndex: Int, attr: String): Node {
        val textField = TextField()
        textField.text = attr
        return textField
    }

    override fun getCurrentAttributeValue(valueNode: Node): String {
        val valField = valueNode as TextField?
        return valField!!.text
    }

    override val initialAttribute: String
        get() = ""

    override fun isSameValue(valA: String, valB: String): Boolean {
        return valA == valB
    }
}